crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -301,3 +355,48 @@ where
         }
     }
 }
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Handle the epilogue error during the note consumption check in the `can_consume` method.
+///
+/// The goal of this helper function is to handle the cases where the account couldn't consume the
+/// note because of some epilogue check failure, e.g. absence of the authenticator.
+fn handle_epilogue_error(epilogue_error: TransactionExecutorError) -> NoteConsumptionStatus {
+    match epilogue_error {
+        // `Unauthorized` is returned for the multisig accounts if the
+        // transaction doesn't have enough signatures.
+        TransactionExecutorError::Unauthorized(_)
+        // `MissingAuthenticator` is returned for the account with the basic
+        // auth if the authenticator was not provided to the executor
+        // (UnreachableAuth).
+        | TransactionExecutorError::MissingAuthenticator => {
+            // Both these cases signal that there is a probability that the provided note could be
+            // consumed if the authentication is provided.
+            NoteConsumptionStatus::UnconsumableWithoutAuthorization
+        },
+        // TODO: apply additional checks to get the verbose error reason
+        _ => NoteConsumptionStatus::Undetermined,
+    }
+}
+
+// HELPER STRUCTURES
+// ================================================================================================
+
+#[derive(Debug, PartialEq)]
+pub enum NoteConsumptionStatus {
```

Comment:
Let's add a doc comment here explaining that this status describes if a note can be consumed under specific conditions.